src/Symfony/Component/String/Slugger/AsciiSlugger.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\String\Slugger;
+
+use Symfony\Component\String\AbstractUnicodeString;
+use Symfony\Component\String\GraphemeString;
+use Symfony\Contracts\Translation\LocaleAwareInterface;
+
+/**
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * @experimental in 5.0
+ */
+class AsciiSlugger implements SluggerInterface, LocaleAwareInterface
+{
+    private const LOCALE_TO_TRANSLITERATOR_ID = [
+        'am' => 'Amharic-Latin',
+        'ar' => 'Arabic-Latin',
+        'az' => 'Azerbaijani-Latin',
+        'be' => 'Belarusian-Latin',
+        'bg' => 'Bulgarian-Latin',
+        'bn' => 'Bengali-Latin',
+        'de' => 'de-ASCII',
+        'el' => 'Greek-Latin',
+        'fa' => 'Persian-Latin',
+        'he' => 'Hebrew-Latin',
+        'hy' => 'Armenian-Latin',
+        'ka' => 'Georgian-Latin',
+        'kk' => 'Kazakh-Latin',
+        'ky' => 'Kirghiz-Latin',
+        'ko' => 'Korean-Latin',
+        'mk' => 'Macedonian-Latin',
+        'mn' => 'Mongolian-Latin',
+        'or' => 'Oriya-Latin',
+        'ps' => 'Pashto-Latin',
+        'ru' => 'Russian-Latin',
+        'sr' => 'Serbian-Latin',
+        'sr_Cyrl' => 'Serbian-Latin',
+        'th' => 'Thai-Latin',
+        'tk' => 'Turkmen-Latin',
+        'uk' => 'Ukrainian-Latin',
+        'uz' => 'Uzbek-Latin',
+        'zh' => 'Han-Latin',
+    ];
+
+    private $defaultLocale;
+
+    /**
+     * Cache of transliterators per locale.
+     *
+     * @var \Transliterator[]
+     */
+    private $transliterators = [];
+
+    public function __construct(string $defaultLocale = null)
+    {
+        $this->defaultLocale = $defaultLocale;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function setLocale($locale)
+    {
+        $this->defaultLocale = $locale;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getLocale()
+    {
+        return $this->defaultLocale;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function slug(string $string, string $separator = '-', string $locale = null): AbstractUnicodeString
+    {
+        $locale = $locale ?? $this->defaultLocale;
+
+        $transliterator = [];
+        if ('de' === $locale || 0 === strpos($locale, 'de_')) {
```

Comment:
For your consideration: I read this code and I don't understand why we need to introduce this exception for the `de` locale. Maybe add a very brief comment explaining why?

Reply:
I did have a comment then removed it, I'll add it back :) . Basically, transliterating German is simple enough that we can provide a polyfill (which is integrated into the ascii() method) and thus we do not need the transliterator.